test/jdk/javax/sound/midi/Sequencer/MetaCallback.java

```diff
@@ -61,6 +64,8 @@ static ShortMessage MidiMsg3(int a, int b, int c) {
 
     int metaCount = 0;
     boolean finished = false;
+    // On M1 Mac sometimes system notifies listener about the same message twice
+    static List<MetaMessage> received = Collections.synchronizedList(new ArrayList<>());
 
```

Comment:
Is this our bug ?

Reply:
it really looks like a bug, but before workaround the bug, it probably would be better to fix threading issues in the test. One of the issues is that the flags are updated in the "meta()" on one thread and checked on another thread.

Reply:
> One of the issues is that the flags are updated in the "meta()" on one thread and checked on another thread.

I can make counter and boolean flag volatile but looking at the logic of the updates i do not think it is a problem since both counter and flag updates happens on the same thread and we do not access counter on another thread until the flag is changed.

Reply:
>  we do not access counter on another thread until the flag is changed.
But we access it on a different thread w/o any synchronization, which we should add.